pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -268,6 +338,18 @@ func (r *ReconcileClusterSync) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	if me, err := r.isSyncAssignedToMe(cd, logger); !me || err != nil {
+		if err != nil {
+			logger.WithError(err).Error("failed determining which instance is assigned to sync this cluster")
+			recobsrv.SetOutcome(hivemetrics.ReconcileOutcomeSyncErrored)
```

Comment:
This outcome is making me a little nervous, it looks like the metrics today do not differentiate success / failure, and yet we don't set outcomes for errors anywhere else. I don't want individual reconcile errors needing an outcome each as that will bloat cardinality. I'd like a reconcile error to generically inject success/failure across all controllers when we report it but I'm not quite sure how to do that. We could perhaps dynamically combine the given outcome with "Success/Failure" someday.

I'd propose we drop this outcome for now and defer reporting error outcomes until a future PR. For now outcome = the action we tried to take, not an indicator of success/failure.

Reply:
Should I just remove this SetOutcome then?

From the code, I believe that would mean we'd get ReconcileOutcomeNoOp for this error path.

I'm fine with that, just making sure that's what you're intending here, or if there's a different outcome you'd prefer.

Reply:
Yeah no-op for now. Thanks.